core/src/main/java/org/apache/carbondata/scan/executor/impl/AbstractQueryExecutor.java

```diff
@@ -470,6 +477,7 @@ private AggregatorInfo getAggregatorInfoForBlock(QueryModel queryModel,
    */
   @Override public void finish() throws QueryExecutionException {
     CarbonUtil.clearBlockCache(queryProperties.dataBlocks);
+    queryIterator.close();
```

Comment:
queryIterator is assigned in subclass?
better to add a if check here for null value